### PR TITLE
Implemented Single Sided Staking

### PR DESCRIPTION
**Pull request by specialist0x**
# xTRI Staking/Unstaking
- Pasted `TRIBAR_ABI` using ABI from trisolaris-core commit `6d710b77ea42c556dbbd428855348efe7bd79b39`
- Added menu button and (un)staking panel
- Created UI and hooks to interact with tribar contract

~~`TriBar` contract has been set as `ZERO_ADDRESS` as a placeholder in this diff.~~
~~Do not merge in to master until this is replaced!~~
Update 12/1: Production TriBar contract has been added (0x802119e4e253D5C19aA06A5d567C5a41596D6803)

# Testing
Tested by checking out `trisolaris-core` on branch `b/xTRI`.
Deploying to mainnet using this script:

```
require('dotenv').config();
const {ethers} = require("hardhat");

async function main() {
    const specialist0x_dev_address = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';

    const Tri = await ethers.getContractFactory("Tri");
    const tri = await Tri.deploy(minter.address);
    await tri.deployed();
    
    const TriBar = await ethers.getContractFactory("TriBar");
    const triBar = await TriBar.deploy(tri.address);
    await triBar.deployed();

    console.log("tri address: ", tri.address);
    console.log("tribar address: ", triBar.address);
    
    let response;
    response = await tri.connect(minter).mint(specialist0x_dev_address, "10000000000000000000"); // this is 10
    console.log('response: ', response);
}

main()
    .then(() => process.exit(0))
    .catch(error => {
        console.error(error);
        process.exit(1);
    });
```

Ran with `yarn hardhat run scripts/test.js --network aurora`:
```
No need to generate any newer typings.
tri address:  0x58Fe52FE3fc1b2caA446dB0Fff9D24e308b03187
tribar address:  0xDb98570C72cc6FDEBFAf934f8d17fA95Ce380e4F
done:  {
  hash: '0x8194ba84a444cce950b8e07434a2ba0785a9bf2249c5f3be94b18a8a1eb3d10b',
  type: 0,
  accessList: null,
  blockHash: '0xc54ef846f823cec78eaa393d6ab4cce2e77fdc95f3e5c802384acc9c75d0a1a7',
  blockNumber: 53995092,
  transactionIndex: 0,
  confirmations: 1,
  from: '0xC5bA85ad8Bc9347f1a22e98a264523024E5Cb9DC',
  gasPrice: BigNumber { _hex: '0x00', _isBigNumber: true },
  gasLimit: BigNumber { _hex: '0x6691b7', _isBigNumber: true },
  to: '0x58Fe52FE3fc1b2caA446dB0Fff9D24e308b03187',
  value: BigNumber { _hex: '0x00', _isBigNumber: true },
  nonce: 1,
  data: '0x40c10f19000000000000000000000000f39fd6e51aad88f6f4ce6ab8827279cfffb922660000000000000000000000000000000000000000000000008ac7230489e80000',
  r: '0xa451333fd959d4d545e089f733ec8ecfc7772a550c2b57a1097a57593015eb12',
  s: '0x22d6a5bc5fa461baf16dd24f0ba461a43c4694bf3095b9b18cb827826767127d',
  v: 2626323143,
  creates: null,
  chainId: 1313161554,
  wait: [Function (anonymous)]
}
```

To test:
- Updated `src/constants/index.ts:95` to use `0x58Fe52FE3fc1b2caA446dB0Fff9D24e308b03187`
- Updated `src/constants/index.ts:107` to use `0xDb98570C72cc6FDEBFAf934f8d17fA95Ce380e4F`

- Staked 1TRI
  - https://explorer.mainnet.aurora.dev/tx/0xdef27c0a7df6d62f01a07f07022b6f164ec2475cbff9aebab11fd59e936a1775/token-transfers
- Staked TRI using `MAX`
  - https://explorer.mainnet.aurora.dev/tx/0xfd7137ecd75888d50e6a3bfc20e38960e306aecd9552a05c99a3ae9e73820d32/token-transfers
- Unstaked 1TRI
  - https://explorer.mainnet.aurora.dev/tx/0x037a44c6de473f23e8a805afa9ac0ddb7f6e80d7dab45b0e9630b7c2a320dd1a/token-transfers
- Unstaked MAX
  - https://explorer.mainnet.aurora.dev/tx/0x0cb411f6cedd4c7a09cebcd3a2c34d85ebeba7a13d501485fc108239ed02796a/token-transfers

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -99,6 +99,19 @@ export const TRI: { [chainId in ChainId]: Token } = {
   )
 }
 
+export const XTRI: { [chainId in ChainId]: Token } = {
+  [ChainId.FUJI]: new Token(ChainId.FUJI, ZERO_ADDRESS, 18, 'xTRI', 'TriBar Token'),
+  [ChainId.AVALANCHE]: new Token(ChainId.AVALANCHE, ZERO_ADDRESS, 18, 'xTRI', 'TriBar Token'),
+  [ChainId.POLYGON]: new Token(ChainId.POLYGON, ZERO_ADDRESS, 18, 'xTRI', 'TriBar Token'),
+  [ChainId.AURORA]: new Token(
+    ChainId.AURORA,
+    '0x802119e4e253D5C19aA06A5d567C5a41596D6803',
```
Comment (baboobhaiya) [inline src/constants/index.ts:108]: can we move this ro the sdk repo?
@0xdon

File: src/pages/StakeTri/index.tsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        <DataCard>
+          <CardBGImage />
+          <CardNoise />
+          <CardSection>
+            <AutoColumn gap="md">
+              <RowBetween>
+                <TYPE.white fontWeight={600}>Earn more TRI</TYPE.white>
+              </RowBetween>
+              <RowBetween>
+                <TYPE.white fontSize={14}>For every swap on the exchange on every chain, XX% of the swap fees are distributed as TRI
```
Comment (0xdon) [inline src/pages/StakeTri/index.tsx:214]: Can replace text segment with:
"0.05% of every trade on the Trisolaris DEX will be used to buy back TRI and distributed to TRI stakers proportional to your share of the staking pool. When you stake TRI you receive xTRI. Your xTRI is continuously compounding and when you unstake you will receive all of your originally deposited TRI as well as additional earned."

File: src/constants/index.ts (hunk 1 above)
Comment (0xdon) [reply src/constants/index.ts:108]: Deployed - please import the latest version of the trisolaris sdk into package.json @specialist0x. It's version 1.6.0 - https://www.npmjs.com/package/@trisolaris/sdk. From the SDK you can import XTRI. And remove it from the constants file here.

Comment (specialist0x) [reply src/constants/index.ts:108]: 👌🏽

Comment (specialist0x) [reply src/constants/index.ts:108]: Adding update based on offline chat:

- Attempted removing this, but wasn't able to import from `@trisolaris/sdk`
- Left this change in place to ensure stability

**Review by 0xchain1 — APPROVED** [review]

**Pull request merged**